### PR TITLE
Create or update baselines to match 2.0

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Comment by smitpatel** [comment]
Why is this required? EF Core already has baselines generated at the time of 2.0 release.

**Comment by ryanbrandenburg** [comment]
As mentioned in the other PR, we're updating all of the baselines to the new patch releases of 2.0.X. That said, we may have changed more than was needed here because of what was discussed [here](https://github.com/aspnet/Hosting/pull/1351). So I won't be merging this quite yet.

**Comment by ryanbrandenburg** [comment]
:up::date: @Eilon there seem to be a number of new breaking changes in here that will need your review.

**Comment by Eilon** [comment]
@smitpatel why do you think this isn't needed? It looks like this is properly capturing many breaking changes. The breaking changes might be deliberate or expected, but we need to make sure of that.

**Comment by smitpatel** [comment]
@Eilon - EF Core (and even SQLite repo) updated all baselines as part of 2.0 release (captured in this PR https://github.com/aspnet/EntityFrameworkCore/pull/9135). The rationale behind that was codebase of "2.0" tag does not contain any breaking changes since it is major release. Since then we have been capturing all the breaking changes and as a team have discussed whether to take the breaking change or implement in non-breaking way.
@ajcvickers & @divega could add more to the process of that. At the least, *Dependencies objects are created so that we can add services to public types without breaking providers.

As for breaking changes in this PR,
I looked at the first one i found in the code in EFCore.Relational project
```
+    { 
+      "TypeId": "public class Microsoft.EntityFrameworkCore.Metadata.RelationalIndexAnnotations : Microsoft.EntityFrameworkCore.Metadata.IRelationalIndexAnnotations", 
+      "MemberId": "public static System.String GetConfiguredColumnType(this Microsoft.EntityFrameworkCore.Metadata.IProperty property)", 
+      "Kind": "Removal" 
+    }
```
I could not find that API ever existing in that type, not in release/2.1,2.0 or 1.1 branch.
Further since we already updated baselines the changes in baselines also look somewhat confusing.

Either the tool has changed a lot from the time we release 2.0 (though surprizing it is still working with older baselines) or there is something which went wrong with generating baselines in this PR.

I give feedback on this PR as "not needed" since we have already done work to update baselines and evaluate breaking changes. All our breaking changes are accurate to mention only breaking changes since 2.0 release.

**Comment by Eilon** [comment]
This one:

```
+    { 
+      "TypeId": "public class Microsoft.EntityFrameworkCore.Metadata.RelationalIndexAnnotations : Microsoft.EntityFrameworkCore.Metadata.IRelationalIndexAnnotations", 
+      "MemberId": "public static System.String GetConfiguredColumnType(this Microsoft.EntityFrameworkCore.Metadata.IProperty property)", 
+      "Kind": "Removal" 
+    }
```

Looks like a bug in the tool. It's clearly an extension method in some other type, that's in `.Internal`: https://github.com/aspnet/EntityFrameworkCore/blob/8a5da5124aabb473ebf24d25094a2f9d40925c98/src/EFCore.Relational/Internal/RelationalPropertyExtensions.cs#L32

(WHich means it should show up in the baseline, but not as a breaking change, and not in this manner.)

But aside from any bugs in the tool, it is not incorrect to update baselines in the major/minor based on the latest patch, because the breaking change rules still apply in exactly the same way. I agree that in general it isn't necessary, but we decided to re-baseline everything on the latest patch for consistency because in lots of repos the baselines were completely wrong.

**Comment by smitpatel** [comment]
It is not incorrect but it points to another issue - what is the use-case of the tool precisely?
- Since 2.0.0 is major version release, it should not have any breaking changes so that particular tag should have up-to-date baselines. (same goes for any other major release)
- In a way breaking changes files in 2.x.x release will capture what are the breaking changes introduced in that release. Updating the baselines we lose that data.
- If we don't update baselines for every 2.x.x release (including patches), that means we are not capturing any new public surface added in code which we may remove later.

As for the baselines being on latest patch, EF Core always merges commit from patch to dev. That includes breaking changes in patch also included dev branch. As commented on SQLite repo, we decided not to update baselines for every patch release. i.e. We decided to ignore point 3 above. Then, does it matter in terms of consistency if EF Core baselines are 2.0.0 & not 2.0.2 ? Given that these files are read exclusively by a tool & humans. Also EF Core had breaking changes in 2.0.1 patch.

https://github.com/aspnet/Universe/issues/524 also discusses that it was done for some high traffic repos. And remaining work should update where it hasn't been done.  Consistency matter went bit too far here. Due to breaking changes in patch (and bugs in tool/generation), now we need to investigate what is changing here & why.

I have nothing against the PR. Its management's call if we should spend time/effort on already working thing.

**Comment by ryanbrandenburg** [comment]
@ajcvickers do we want to track breaking changes for src/EFCore.Specification.Tests/baseline.netcore.json? @Eilon and I think no but wanted to make sure.

File: src/EFCore.Relational.Design.Specification.Tests/baseline.netcore.json [hunk 1]
```diff
@@ -0,0 +1,1392 @@
+{
```
Comment (ryanbrandenburg) [inline src/EFCore.Relational.Design.Specification.Tests/baseline.netcore.json]: This package doesn't exist anymore, not to self to remove it and the netframework version.

**Comment by ajcvickers** [comment]
@ryanbrandenburg Breaking changes in the specs tests are fine.

**Comment by Eilon** [comment]
@ryanbrandenburg ok let's remove the Spec baselines and disable ApiCheck for them, then check in what's left.

**Comment by ryanbrandenburg** [comment]
Removed Spec baselines, remove extra files added by accident. Will merge once tests pass.

**Comment by AndriySvyryd** [comment]
@ryanbrandenburg Most of the changes to breakingchanges aren't real.
This wasn't merged into dev as the incorrect entries might hide potential real breaking changes. Please submit a corrected PR for dev.
Also Relational Specification Tests shouldn't have a baseline.

**Comment by Eilon** [comment]
@AndriySvyryd we're aware that the breaking changes list has some bogus stuff in it due to some known bugs in the tool. We have bugs filed for those and will address during MQ.

**Pull request merged**